afm/server.py

```diff
@@ -161,11 +165,18 @@ def do_get(self, context, ticket: fl.Ticket):
 
     def do_put(self, context, descriptor, reader, writer):
         asset_info = json.loads(descriptor.command)
+        logger.trace(asset_info)
         logger.info('writing dataset',
                     extra={DataSetID: asset_info['asset'],
                            ForUser: True})
+        # default write mode is overwrite
+        write_mode = 'overwrite'
+        if 'write_mode' in asset_info:
+           write_mode = asset_info['write_mode']
+        if write_mode not in ['append', 'overwrite']:
+           raise ValueError("Unsupported write mode type: {}".format(write_mode))
         with Config(self.config_path) as config:
-            asset = asset_from_config(config, asset_info['asset'], capability="write")
+            asset = asset_from_config(config, asset_info['asset'], capability="write", write_mode=write_mode)
             self._write_asset(asset, reader)
```

Comment:
We can add a parameter for `write_mode` to `_write_asset()` instead of adding a field to `Asset` class. I think the `write_mode` is not part of the `Asset`.

Reply:
Fixed. Thanks